### DISCUSSION TITLE
Dynamic path handler folder name

### DISCUSSION THREAD

**Discussion by cd-slash** [beginner-corp/begin-community#96] [Support] [answered]
I am working on my first Begin app and really enjoying it so far :-)

I created a route at `/foo` that worked well with my handler located at `/src/http/get-foo` and the associated entry in my package.json, per the docs. When updating this to be a dynamic route at `/foo/:bar` with package.json entry `["get", "/foo/:bar]` I get an error:

    Could not find Lambda handler at: /my/local/path/src/http/get-foo-000bar/index.js

I can of course work around this by renaming my handler folder to `get-foo-000bar` but this seems rather unintuitive and I can't see anywhere in the [reference docs](https://docs.begin.com/en/http-functions/provisioning#using-url-parameters-to-create-dynamic-paths) that tells me how to name the handler folders for dynamic routes. Am I missing it somewhere?

**Comment by cd-slash** [comment, marked as answer]
I now see that there's an example of a dynamic route in the [project strcture](https://docs.begin.com/en/getting-started/project-structure#srchttp) guide that does indeed show the dynamic part of the URL prefixed by `000` so it seems this is indeed the correct way to handle a route like this. It may be helpful to tweak the docs slightly to make this clear (and perhaps explain why the `000` prefix is required).

**Comment by brianleroux** [comment]
it is not required! you can set custom source paths; sorry these docs aren't great. you can reference the architect docs on this here https://arc.codes/docs/en/guides/developer-experience/custom-source-paths
